controllers/awsmachine_controller.go

```diff
@@ -415,13 +415,13 @@ func (r *AWSMachineReconciler) AWSClusterToAWSMachines(o handler.MapObject) []ct
 	}
 
 	labels := map[string]string{clusterv1.MachineClusterLabelName: c.Name}
-	machineList := &infrav1.AWSMachineList{}
+	machineList := &clusterv1.MachineList{}
 	if err := r.List(context.Background(), machineList, client.InNamespace(c.Namespace), client.MatchingLabels(labels)); err != nil {
-		r.Log.Error(err, "failed to list AWSMachines", "AWSCluster", c.Name, "Namespace", c.Namespace)
+		r.Log.Error(err, "failed to list Machines", "AWSCluster", c.Name, "Namespace", c.Namespace)
 		return nil
 	}
 	for _, m := range machineList.Items {
-		name := client.ObjectKey{Namespace: m.Namespace, Name: m.Name}
+		name := client.ObjectKey{Namespace: m.Namespace, Name: 	m.Spec.InfrastructureRef.Name}
```

Comment:
I'm wondering if we need to guard against the InfrastructureRef not being set yet here.

Reply:
I thought about that yesterday after i turned my computer off. +1, will update